### PR TITLE
Expose ZEPHYR_TOOLCHAIN_VARIANT to Kconfig to support openisa_rv32m1

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Comment by galak** [comment]
Part of my reasoning for this is this will allow us to build test / regress the vega board and openisa_rv32m1 soc/drivers with the standard Zephyr toolchain (once 0.10.0) is official.

**Review by mbolivar — APPROVED** [review]
I think this is a reasonable thing to do. Users of toolchains with ISA extensions are going to be setting ZEPHYR_TOOLCHAIN_VARIANT to cross-compile in practice.

File: soc/riscv32/openisa_rv32m1/Kconfig.defconfig [hunk 1]
```diff
@@ -29,6 +29,7 @@ config XIP
 
 config RISCV_GENERIC_TOOLCHAIN
 	bool
+	default y if "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr"
 	default n
```
Comment (ulfalizer) [inline soc/riscv32/openisa_rv32m1/Kconfig.defconfig]: Another variant:

```
config RISCV_GENERIC_TOOLCHAIN
	bool
	default "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr"
```

Or this:

```
config RISCV_GENERIC_TOOLCHAIN
	def_bool "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr"
```

The value for `default`, etc., can be any expression.

**Review by ulfalizer — APPROVED** [review]

**Comment by galak** [comment]
@SebastianBoe can you take a look

**Review by carlescufi — APPROVED** [review]

**Review by SebastianBoe — COMMENTED** [review]
This is broken. You forgot 

`set(ENV{ZEPHYR_TOOLCHAIN_VARIANT} ${ZEPHYR_TOOLCHAIN_VARIANT})
`

**Comment by galak** [comment]
> `set(ENV{ZEPHYR_TOOLCHAIN_VARIANT} ${ZEPHYR_TOOLCHAIN_VARIANT}) `

I think its not needed since its already coming from the env.

**Comment by SebastianBoe** [comment]
Aha, then it is pretty close.

But it is valid (and has precedence) to set ZEPHYR_TOOLCHAIN_VARIANT as cmake -DZEPHYR_TOOLCHAIN_VARIANT=gnuarmemb.

**Pull request merged**